File: Fragment1/app/src/main/java/com/example/fragment_1/AnimalAdapter.java
package com.example.fragment_1;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class AnimalAdapter extends RecyclerView.Adapter<AnimalViewHolder> {
    private List<AnimalModel> animalModelList = new ArrayList<>();

    public AnimalAdapter(List<AnimalModel> animalModelList) {
        this.animalModelList = animalModelList;
    }

    @NonNull
    @Override
    public AnimalViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.animal_item_layout, parent, false);
        return new AnimalViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AnimalViewHolder holder, int position) {
        AnimalModel animalModel = animalModelList.get(position);
        holder.setData(animalModel);
    }

    @Override
    public int getItemCount() {
        return animalModelList.size();
    }
}

File: Kotlin Projects/AnimalsprofileinaFragment/app/src/main/java/com/example/animalsprofileinafragment/AnimalAdapter.kt
package com.example.animalsprofileinafragment

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class AnimalAdapter(val animalModelList: List<AnimalModel>) : RecyclerView.Adapter<AnimalViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AnimalViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.animal_item_layout, parent, false)
        return AnimalViewHolder(view)
    }

    override fun onBindViewHolder(holder: AnimalViewHolder, position: Int) {
        val animalModel: AnimalModel = animalModelList[position]
        holder.setAnimalData(animalModel)
    }

    override fun getItemCount(): Int {
        return animalModelList.size
    }
}